### PR TITLE
Adds Summon Party Member to the wizard's spellbook

### PR DESCRIPTION
**Pull request by datlo**
## What Does This PR Do

This PR allows wizards to summon non-magical allies to assist them. 1 spell point cost, 3 buys max.

With the Wizard Federation's rapid expansion, they are struggling to meet hiring quotas, and have 'convinced' Nanotrasen's HR department to assist in hiring party members.

All party members come with runed armor variants which increase defenses to wizard robes tier (or better) and remove slowdown, but cannot be used to cast spells. They do not have ID, PDAs, or departmental headsets.

Code-wise they are a subtype of apprentices so the round will keep going while they are alive even if the wizard is dead.

This PR also refactors the wizard apprentice contract to show ghosts exactly what roles they are rolling for (Honk wizard apprentice, Barbarian party member, etc)

You can pick one of 3 classes for your summon : 
Barbarian 
image
The red spray bottle is loaded with 250 units of Chlorine Trifluorine which will light stuff and people on fire.

Cleric 

image
Comes with a surgery arm implant, the pill bottle and patch pack are loaded with a variety of medecines, including hydrocodone for surgery and lazarus reagent pills to revive fallen team mates.

Fighter

image

The amount of gear given to each class will likely be tweaked working with the balance team.

## Why It's Good For The Game

More summoning options means the antag fun can be shared to more people and could possibly extend wizard rounds a bit further. These summons lack the wizard/apprentice mobility spells, so they emphasize teamwork and a more direct fighting approach as opposed to standard hit and run wizard tactics.

## Images of changes

The boys
image

New TGUI window for the contract
image

## Testing

Tested both the new contract and the refactor apprentice contract with two accounts, checked that all items are there and every change works as intended.

<hr>

### Declaration
- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/feature_freeze/CODE_OF_CONDUCT.md#currently-changes-to-the-following-types-of-content-reuires-pre-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.

image
image
image
image

<hr>

## Changelog
:cl:
add: Wizards can now summon up to 3 party members, Cleric, Fighter, or Barbarian, for 1 spell point each. Party members do not have spells but are fully equipped with doctor, security, or axe-crazy atmos tech gear respectively.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/wizard/artefact.dm [hunk 1]
```diff
@@ -4,19 +4,23 @@
...
-	icon_state ="scroll2"
+	icon_state = "scroll2"
 	throw_speed = 1
 	throw_range = 5
 	w_class = WEIGHT_CLASS_TINY
 	var/used = FALSE
+	var/tgui_window = "WizardApprenticeContract"
+	var/datum_to_spawn = /datum/antagonist/wizard/apprentice
+	var/poll_question = "Do you want to play as the wizard apprentice?"
+	var/poll_role = "wizard apprentice"
```
Comment (Sirryan2002) [inline code/game/gamemodes/wizard/artefact.dm:15]: Document these variables

File: code/game/gamemodes/wizard/artefact.dm [hunk 2]
```diff
@@ -64,11 +71,20 @@
...
 	if(used)
-		to_chat(user, "<span class='warning'>You've already summoned an apprentice or you are in process of summoning one.</span>")
+		to_chat(user, "<span class='warning'>You've already used this scroll or you are in process of using it.</span>")
 		return
 
 	ui_interact(user)
 
+/////////Party Member Contract//////////
+
+/obj/item/contract/partymember
```
Comment (Sirryan2002) [inline code/game/gamemodes/wizard/artefact.dm:81]: document this type!

File: code/game/gamemodes/wizard/spellbook.dm [hunk 3]
```diff
@@ -634,7 +634,16 @@
 	item_path = /obj/item/contract
 	category = "Summons"
 	limit = 1
-	is_ragin_restricted = TRUE //We have enough wizards already! Sheesh!
+	is_ragin_restricted = TRUE // We have enough wizards already! Sheesh!
+
+/datum/spellbook_entry/item/summonpartymember
+	name = "Summon Party Member"
+	desc = "Reinforce your party with your choice of Cleric, Barbarian, or Warrior! Due to lack of staffing, we've 'convinced' Nanotrasen's HR department to help us out. \
```
Comment (Sirryan2002) [inline code/game/gamemodes/wizard/spellbook.dm:641]: what is the reasoning behind the HR department message?

File: code/game/objects/items/weapons/storage/belt.dm [hunk 4]
```diff
@@ -182,6 +182,16 @@
...
 
+/obj/item/storage/belt/utility/barbarian/populate_contents()
+	new /obj/item/screwdriver(src)
+	new /obj/item/wrench(src)
+	new /obj/item/weldingtool(src)
+	new /obj/item/crowbar/power(src)
+	new /obj/item/multitool(src)
+	new /obj/item/extinguisher/mini(src)
+	new /obj/item/stack/cable_coil/random(src, 30)
+	update_icon()
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/storage/belt.dm:193]: maybe this stuff would be better defined with a wizard specific outfit file?

File: code/modules/antagonists/wizard/datum_party_member.dm [hunk 5]
```diff
@@ -0,0 +1,112 @@
+/datum/antagonist/wizard/apprentice/partymember
```
Comment (Sirryan2002) [inline code/modules/antagonists/wizard/datum_party_member.dm:1]: document this type!

File: code/modules/antagonists/wizard/datum_party_member.dm [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+	name = "Wizard Party Member"
+	greeting_text = "<span class='danger'>You have been recruited as the Wizard's party member! \
+		You are bound by magic contract to follow their orders and help them in accomplishing their goals.</span>"
+
+/datum/antagonist/wizard/apprentice/greet()
+	var/list/messages = list()
+	. = messages
```
Comment (Sirryan2002) [inline code/modules/antagonists/wizard/datum_party_member.dm:8]: what is this point of this line?

File: code/modules/antagonists/wizard/datum_party_member.dm [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+	switch(class_type)
+		if("Cleric")
+			new_wiz.equipOutfit(/datum/outfit/wizard_summon/cleric)
+			messages += "<b>You are equipped with an armored labcoat, a telescopic baton, a variety of medical supplies and a surgery tools implant. Keep your party healthy!</b>"
+		if("Barbarian")
+			new_wiz.equipOutfit(/datum/outfit/wizard_summon/barbarian)
+			messages += "<b>You are equipped with fully fireproof armor, a fire axe, a whetstone, and a short range fire spray. Sharpen your axe, and charge at your enemies!</b>"
+		if("Fighter")
+			new_wiz.equipOutfit(/datum/outfit/wizard_summon/fighter)
+			messages += "<b>You are equipped with reinforced riot armor and a variety of security gear including a stun baton, disabler and riot shotgun. Get in the fight!</b>"
```
Comment (Sirryan2002) [inline code/modules/antagonists/wizard/datum_party_member.dm:34]: convert class types into defines please

File: tgui/packages/tgui/interfaces/WizardPartyMemberContract.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+          <p>
+            If you are unable to summon a member, you can feed the contract back to the spellbook to refund your point.
+          </p>
+          {used ? (
+            <Box bold color="red">
+              You&apos;ve already summoned an ally or you are in process of summoning one.
+            </Box>
+          ) : (
+            ''
+          )}
```
Comment (Sirryan2002) [inline tgui/packages/tgui/interfaces/WizardPartyMemberContract.js:22]:
```suggestion
          {Boolean(used) &&
            <Box bold color="red">
              You&apos;ve already summoned an ally or you are in process of summoning one.
            </Box>
          }
```

File: tgui/packages/tgui/interfaces/WizardPartyMemberContract.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+              <Button content="Select" disabled={used} onClick={() => act('Barbarian')} />
+            </LabeledList.Item>
+            <LabeledList.Divider />
+            <LabeledList.Item label="Fighter">
+              You will summon a Fighter.
+              <br />
+              They are equipped with a reinforced riot suit, a riot shotgun, spare non-lethal ammo, a disabler, a
+              stunbaton and a variety of security gear.
+              <br />
+              <Button content="Select" disabled={used} onClick={() => act('Fighter')} />
```
Comment (Sirryan2002) [inline tgui/packages/tgui/interfaces/WizardPartyMemberContract.js:47]: where is this new `act()` being referenced?

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/wizard/artefact.dm [hunk 10]
```diff
@@ -64,11 +71,20 @@
...
 
 	if(used)
-		to_chat(user, "<span class='warning'>You've already summoned an apprentice or you are in process of summoning one.</span>")
+		to_chat(user, "<span class='warning'>You've already used this scroll or you are in process of using it.</span>")
 		return
 
 	ui_interact(user)
 
+/////////Party Member Contract//////////
+
```
Comment (Burzah) [inline code/game/gamemodes/wizard/artefact.dm:80]:
```suggestion
// Party Member Contract
```

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 11]
```diff
@@ -435,3 +446,11 @@
 	for(var/I in 1 to 5)
 		new /obj/item/reagent_containers/patch/silver_sulf/small(src)
 		new /obj/item/reagent_containers/patch/styptic/small(src)
+
+/obj/item/storage/pill_bottle/patch_pack/cleric
+	wrapper_color = COLOR_BLUE_LIGHT
+
+/obj/item/storage/pill_bottle/patch_pack/cleric/populate_contents()
+	for(var/I in 1 to 5)
```
Comment (Burzah) [inline code/game/objects/items/weapons/storage/firstaid.dm:454]:
```suggestion
	for(var/i in 1 to 5)
```
Nit

File: code/game/gamemodes/wizard/spellbook.dm (hunk 3 above)
Comment (lewcc) [reply code/game/gamemodes/wizard/spellbook.dm:641]: Strikes me as them using some wizard mind control or something.

Or just plain blackmail?

File: code/game/gamemodes/wizard/spellbook.dm [hunk 12]
```diff
@@ -738,11 +747,20 @@
...
 		var/obj/item/contract/contract = O
 		if(contract.used)
-			to_chat(user, "<span class='warning'>The contract has been used, you can't get your points back now!</span>")
-		else
-			to_chat(user, "<span class='notice'>You feed the contract back into the spellbook, refunding your points.</span>")
-			uses+=2
-			qdel(O)
+			to_chat(user, "<span class='warning'>The contract has been used, or is in the process of summoning!</span>")
+			return
+		if(istype(contract,	/obj/item/contract))
```
Comment (lewcc) [inline code/game/gamemodes/wizard/spellbook.dm:752]:
```suggestion
		if(istype(contract))
```

**Comment by datlo** [comment]
Don't have the time to work on this atm but hoping to come back to it!

**Pull request closed without merging**